### PR TITLE
Adding Phobos station

### PR DESCRIPTION
**Pull request by Starbuckss14**
## About the PR

This is DarkDan's map Phobos! A mid sized map featuring a asteroid theme!

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

add: added Phobos station!

### CODE REVIEW

**Comment by Mangohydra** [comment]
best map ever

**Comment by lzk228** [comment]
bagel 3.0??

**Comment by UbaserB** [comment]
feels like a lot of weirdly empty space

**Comment by Starbuckss14** [comment]
> best map ever

Why thank you!

**Comment by Starbuckss14** [comment]
> bagel 3.0??

Lol It kind of looks like that

**Comment by Starbuckss14** [comment]
> feels like a lot of weirdly empty space

sorry what part?

**Comment by IProduceWidgets** [comment]
Are the candles the decorative / always lit variety? It seems like a lot of candles. I'm not against candles, I just noticed the quantity.

Perma having an actual non-joke disposal even though it goes to space seems kinda uhhhhh.

No meteor protection (lattice / grilles) on an already very spaceable map.

Free gun in the bar!?

Clown HOS feels very NRP.

I don't like wall mounts on rock. There's a lot on the north side including lockers/closets and apcs.

I wanna say probably don't map the bananium horn. 

Would like others thoughts on the sauna. I fear it may just be a death box.

Bridge hallway emergency lockers feel shoehorned and probably just need a small recess to sit in.

Why are there syndicate airlocks on the station (in the north)???

I couldn't find where your disposals room is but maybe I'm blind or weh. 

Disclaimer: I did not load the map ingame, and so did not check wiring/power/pipes/atmos/dispos or anything else that would require that.

**Comment by Hmeister-real** [comment]
my thoughts: map looks fun, not many flaws i can point out, but a few are that
1) That Bar revolver should be in a locked Barkeep accessable windoor 1x1 chamber behind the counter, or jusr in the backroom, not open for a tider to rush to round start
2) the map looks extremely spacable, so engi should spawn with a ton more at, the very least, inflatable walls if not more.
3) the moment a syndie/shitter get a pickaxe and an EVA its over.

**Comment by Errant-4** [comment]
Can you please the PR template again. Thanks

File: Resources/Prototypes/Maps/Pools/default.yml [hunk 1]
```diff
@@ -14,3 +14,4 @@
   - Saltern
   - Packed
   - Reach
+  - phobos
```
Comment (Errant-4) [inline Resources/Prototypes/Maps/Pools/default.yml:17]: Literally every other map in this list is capitalised, and in alphabetical order

File: Resources/Prototypes/Maps/phobos.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+- type: gameMap
+  id: phobos
+  mapName: 'phobos'
```
Comment (Errant-4) [inline Resources/Prototypes/Maps/phobos.yml:3]: 'Phobos'

File: Resources/Prototypes/Maps/phobos.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+- type: gameMap
+  id: phobos
+  mapName: 'phobos'
+  mapPath: /Maps/phobos.yml
+  minPlayers: 0
+  maxPlayers: 50
```
Comment (Errant-4) [inline Resources/Prototypes/Maps/phobos.yml:6]: I'm no expert but min 0 max 50 seems like an... odd range? Was it intended like this?

Comment (Starbuckss14) [reply Resources/Prototypes/Maps/phobos.yml:6]: I just kinda picked a number. Any suggestions on making it higher/lower?

File: Resources/Prototypes/Maps/Pools/default.yml (hunk 1 above)
Comment (Starbuckss14) [reply Resources/Prototypes/Maps/Pools/default.yml:17]: I will fix that

File: Resources/Prototypes/Maps/phobos.yml (hunk 3 above)
Comment (Errant-4) [reply Resources/Prototypes/Maps/phobos.yml:6]: I'm not really a mapper, but a map that is targeted for up to 50 players should have a minimum around 15-20 , judging by other map prototypes. Might want to work that out with experienced mappers in Cartography

File: Resources/Prototypes/Maps/Pools/default.yml (hunk 1 above)
Comment (Errant-4) [reply Resources/Prototypes/Maps/Pools/default.yml:17]: For the record, you are supposed to Resolve reviews _after_ you have deployed the fix

**Comment by CMDR-Piboy314** [comment]
image
what the hell is this laser also the fireplace in the wall in the library is pretty horrific

add some lattices around in space and solars out there too, lattices connecting e.g. the bridge thats kind of hanging off there would be nice as well.

adding some medical supply cabinets would be much appreciated and there probably shouldn't be a jug vendor in that hallway to cmo. also the morgue having syndicate airlocks is not a great idea but the syndicate posters and stuff are cool. the sand looks really out of place for the cmo walls.

rd literally cant get out of their room????
image

the chef's bedroom thing is really weird and fireplaces being anywhere but under walls looks pretty horrible imo, also why the hell is there an atmos crab there... in some maints there are way too many candles like south.

this is also pretty horrible, there's a tree covering the maints entrance, so many chairs in a row etc.
image

**Comment by mirrorcult** [comment]
> rd literally cant get out of their room???? image
> 

pretty sure this is just a maprenderer bug, it should be 1 tile up in game

**Comment by UbaserB** [comment]
> > rd literally cant get out of their room???? image
> 
> pretty sure this is just a maprenderer bug, it should be 1 tile up in game

same with the trees

**Comment by Starbuckss14** [comment]
> > rd literally cant get out of their room???? image
> 
> pretty sure this is just a maprenderer bug, it should be 1 tile up in game

Ya its a bug in the render

File: Resources/Prototypes/Maps/phobos.yml (hunk 3 above)
Comment (Starbuckss14) [reply Resources/Prototypes/Maps/phobos.yml]: Map size updated to 65

**Comment by Starbuckss14** [comment]
> image what the hell is this laser also the fireplace in the wall in the library is pretty horrific
> 
> add some lattices around in space and solars out there too, lattices connecting e.g. the bridge thats kind of hanging off there would be nice as well.
> 
> adding some medical supply cabinets would be much appreciated and there probably shouldn't be a jug vendor in that hallway to cmo. also the morgue having syndicate airlocks is not a great idea but the syndicate posters and stuff are cool. the sand looks really out of place for the cmo walls.
> 
> rd literally cant get out of their room???? image
> 
> the chef's bedroom thing is really weird and fireplaces being anywhere but under walls looks pretty horrible imo, also why the hell is there an atmos crab there... in some maints there are way too many candles like south.
> 
> this is also pretty horrible, there's a tree covering the maints entrance, so many chairs in a row etc. image

Laser removed. Med tweaked. Hydro given a pipe system to the chef and chem. Map render bug. Fireplace fixed.

File: Resources/Prototypes/Maps/phobos.yml (hunk 2 above)
Comment (Errant-4) [reply Resources/Prototypes/Maps/phobos.yml]: Okay my mistake for not being more specific. As far as I can see, only this specific string, `mapName`, should be capitalised. Not id, and not the file's name. Look at the other map files. While blindly copying things has it's own problems, when you are making something new then looking at the existing examples is probably still your best starting point, and consistency is good

File: Resources/Prototypes/Maps/phobos.yml (hunk 3 above)
Comment (Errant-4) [reply Resources/Prototypes/Maps/phobos.yml]: I repeat: the issue is that you have configured the map to be an equally valid choice (as far as the game engine is concerned) for 65 players or 4. Do you see the issue with putting 4 players on a map large enough for 65

Comment (Starbuckss14) [reply Resources/Prototypes/Maps/phobos.yml]: OK I understand the issue. I will count up the spawns and leave like 10 spots for wiggle room for greys?

Comment (Errant-4) [reply Resources/Prototypes/Maps/phobos.yml]: Look at a similar sized map to get an idea for individual role slot numbers, but specifically I meant that you need to set a `minPlayers` value so the map does not get picked if there are too few players in the lobby

**Comment by Starbuckss14** [comment]
> Can you please the PR template again. Thanks
> 
> (If it gets merged then the description you provide will also be used on the wiki so it would be helpful to have something)
> 
> Putting a map in rotation also needs to be mentioned in the Changelog

Fixed

**Comment by Starbuckss14** [comment]
Is there a way for me to make a new render of my map so the render is up to date?

**Comment by Errant-4** [comment]
> Is there a way for me to make a new render of my map so the render is up to date?

`dotnet run --project Content.MapRenderer Phobos`

**Comment by Starbuckss14** [comment]
> > Is there a way for me to make a new render of my map so the render is up to date?
> 
> `dotnet run --project Content.MapRenderer Phobos`

Where do I run that command?

**Comment by Errant-4** [comment]
> > > Is there a way for me to make a new render of my map so the render is up to date?
> > 
> > 
> > `dotnet run --project Content.MapRenderer Phobos`
> 
> Where do I run that command?

In any command line interface (that is currently focused on the directory of your project). Your IDE should have a terminal built in, and will already be focused on your project files. Or you can use the one provided by your OS

**Comment by Starbuckss14** [comment]
> > > > Is there a way for me to make a new render of my map so the render is up to date?
> > > 
> > > 
> > > `dotnet run --project Content.MapRenderer Phobos`
> > 
> > 
> > Where do I run that command?
> 
> In any command line interface (that is currently focused on the directory of your project). Your IDE should have a terminal built in, and will already be focused on your project files. Or you can use the one provided by your OS

I am having issues with getting the render to work. Is there any documentation on it?

**Comment by Starbuckss14** [comment]
> dotnet run --project Content.MapRenderer Phobos

I found my issue the map renderer is out of date and wont work because of the newer prototypes. I will take some photos instead

**Comment by Starbuckss14** [comment]
I made some changes based on the suggestions. Below are some images of the map. Let me know what else needs fixing before we are ready to merge.

The maints needed some more hiding spots so I added this new "old bridge" 
Screenshot (43)
Changed the HOS's office and removed the clown mask
Screenshot (44)
Added a pipe system for hydro to send and receive items from the chef and chem 
Screenshot (45)
Screenshot (46)
updated the abandoned syndi shuttle to have medical airlocks
Screenshot (47)

**Comment by MjrLandWhale** [comment]
Here's my feedback for what it's worth:

1. There are far too many space facing walls on the west side which are not reinforced. All other stations (to my knowledge) exclusively use reinforced walls when separating station from space. In my mind this should also apply where the only additional support is something like rocks. As an example near the science department, anything marked in red _should_ be reinforced in my mind
image

2. The inclusion of many vending machines directly in the hallways feels somewhat... clunky to me. My recommendation is to recess these sections one tile where possible. Something like we see in other stations (origin here as an example):
image
You already take this approach on the eastern main hallway, where you elect to go with a 4 wide, using the extra space for chairs, tables, etc. 

3. Further compounding the issues from #1 is the lack of any maintenance-based firelocks to control loss of atmosphere.

I think overall I have major concerns about the survivability of this station from sabotage. With only a welder, wrench, and pickaxe I can:

- break 8 rocks and space most of medical. Also deconstruct 2 normal walls and you've spaced the main hall outside med and the hallway leading to science.
- Deconstruct 2 normal walls to space the main hallway (and maints) by sec & cargo, effectively cutting off security unless they have a suit or run around the entire station.
- Break 2 rocks and space dorms.
- Deconstruct 2 normal walls and plasma flood the entire hallway outside engineering while also venting the station's entire plasma gas supply. (side note: why in the world is the plasma gas hooked directly into distro via a manual valve?)

Unless we have a way of giving nearly everyone hardsuits right off the bat, I fear this station in its current form will result in a rapid, widespread loss of atmosphere. This will have the consequence of killing/injuring far too much of the station without any meaningful way to control or stop it.

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Pull request closed without merging**